app/scripts/controllers/preferences-controller.ts

```diff
@@ -516,47 +615,49 @@ export default class PreferencesController {
   /**
    * A getter for the `preferences` property
    *
-   * @returns {object} A key-boolean map of user-selected preferences.
+   * @returns A map of user-selected preferences.
    */
-  getPreferences() {
+  getPreferences(): Preferences {
     return this.store.getState().preferences;
   }
 
   /**
    * A getter for the `ipfsGateway` property
    *
-   * @returns {string} The current IPFS gateway domain
+   * @returns The current IPFS gateway domain
    */
-  getIpfsGateway() {
+  getIpfsGateway(): string {
     return this.store.getState().ipfsGateway;
   }
 
   /**
    * A setter for the `ipfsGateway` property
    *
-   * @param {string} domain - The new IPFS gateway domain
-   * @returns {Promise<string>} A promise of the update IPFS gateway domain
+   * @param domain - The new IPFS gateway domain
+   * @returns A promise of the update IPFS gateway domain
    */
-  async setIpfsGateway(domain) {
+  async setIpfsGateway(domain: string): Promise<string> {
```

Comment:
I believe this function shouldn't be async in first place, once removed the return type would be string